src/lava/utils/float2fixed.py

```diff
@@ -1,26 +1,719 @@
+        Variable Name key precision, domain, constant, and num_exp_bit.
+
+        Returns
+        -------
+        conv_data : dict
+            Conversion data for Float2Fixed converter in nested dict.
+            {ProcID:
+                {VarName:
+                    {'precision': _,
+                     'domain': _,
+                     'constant: _,
+                     'num_bits_exp':_}}}
+        """
+        # Create dictionary to store conversion data.
+        conv_data = {}
+
+        for p_id, p_var_ports in self.var_ports.items():
+            # Increase depth of dictionary to collect information about Vars of
+            # Process and connections.
+            conv_data[p_id] = {}
+            # Get Process
+            p = self.procs[p_id]
+            # Get ProcessModel.
+            p_model = self.fixed_pt_proc_models[p_id]
+
+            # Check whether process has out connections that are scaled.
+            # If so, check for consistency of this scaling
+            p_impl_shift = False
+            for out in p_var_ports['SourcePort']:
+                # Get target Ports as list.
+                target_ports = p.__getattribute__(out).out_connections
+
+                # Get if port is OutPort
+                outport = (type(p.__getattribute__(out)) is OutPort)
+
+                # Update target Ports. If target Port is Port of hierarchcical
+                # Procass find next Prots of Subprocesses.
+                target_ports = self._update_target_ports(target_ports)
+
+                for t_port in target_ports:
+
+                    # If Outport get name of target Port.
+                    if outport:
+                        t_name = t_port.name
+                    # Else get name of variable targeted by RefPort
+                    else:
+                        t_name = t_port.var.name
+
+                    # Get id of Process hosting target Port.
+                    t_port_p_id = t_port.process.id
+                    # Get ProcessModel of Process hosting target Port.
+                    t_port_p_model = self.fixed_pt_proc_models[t_port_p_id]
+                    # Get LavaPyType of target Port.
+                    t_port_py_type = t_port_p_model.__getattribute__(t_name)
+                    # Get precision information of target Port
+                    try:
+                        t_port_conv_data = t_port_py_type.conversion_data()
+                    except (TypeError, ValueError):
+                        raise ValueError(f'Error in {t_port} of Proc {p_id}')
+                    t_port_impl_shift = t_port_conv_data['implicit_shift']
+                    # If there has been not implicit shift set so far set one.
+                    if not p_impl_shift:
+                        p_impl_shift = t_port_impl_shift
+
+                    # Check for consistency of implicit shifts due to Ports
+                    elif not p_impl_shift == t_port_impl_shift:
```

Comment:
Where could this inconsistent shift come from? It looks like there is a redundant way for the user to set this. Is there any way we may prevent this in general?

Reply:
Say you write you custom adapted neuron model - let's call it PLIF - which has a different implicit shift then the classical LIF implemented in Lava.  
Now you instantiate a `plif` and `lif` process and connect it to the same `dense` process.  
In this case, we would have inconsistent implicit shifts.  
I think in general this is not preventable.